src/stdlib_constants.fypp

```diff
@@ -0,0 +1,64 @@
+#:include "common.fypp"
+#:set KINDS = REAL_KINDS
+module stdlib_constants
+    !! Constants
+    !! ([Specification](../page/specs/stdlib_constants.html))
+    use stdlib_kinds, only: #{for k in KINDS[:-1]}#${k}$, #{endfor}#${KINDS[-1]}$
+    use stdlib_codata, only: SPEED_OF_LIGHT_IN_VACUUM, &
+                             VACUUM_ELECTRIC_PERMITTIVITY, &
+                             VACUUM_MAG_PERMEABILITY, &
+                             PLANCK_CONSTANT, &
+                             NEWTONIAN_CONSTANT_OF_GRAVITATION, &
+                             STANDARD_ACCELERATION_OF_GRAVITY, &
+                             ELEMENTARY_CHARGE, &
+                             MOLAR_GAS_CONSTANT, &
+                             FINE_STRUCTURE_CONSTANT, &
+                             AVOGADRO_CONSTANT, &
+                             BOLTZMANN_CONSTANT, &
+                             STEFAN_BOLTZMANN_CONSTANT, &
+                             WIEN_WAVELENGTH_DISPLACEMENT_LAW_CONSTANT, &
+                             RYDBERG_CONSTANT, &
+                             ELECTRON_MASS, &
+                             PROTON_MASS, &
+                             NEUTRON_MASS, &
+                             ATOMIC_MASS_CONSTANT
+    private
+
+    ! mathematical constants
+    #:for k in KINDS
+    real(${k}$), parameter, public :: PI_${k}$ = acos(-1.0_${k}$) !! PI
+    #:endfor
+    
+    ! Physical constants
+    real(dp), parameter, public :: c = SPEED_OF_LIGHT_IN_VACUUM%value !! Speed of light in vacuum
```

Comment:
I would be careful with one letter variables or constants. There are many physical constants and we would not be able to provide their short forms in Fortran nicely because the same letter might be used for several constants. Also, because of the case insensitivity of the language these names might bring surprises for users.

Reply:
The idea was to mimic the constants module as it is done in Scipy. Nonetheless, I'm aware of the potential problem with one letter constants. If everybody agrees I can remove them.

Reply:
Instead of removing them, could we think to add a prefix, e.g.,?
```fortran 
... stdlib_pi = ...
... stdlib_c = ...
```

Reply:
When using numeric constants, the shorter their names the better imho. Because they're inside a module that only contains them, I think they're mostly harmless. In case one has overlapping names, they can always rename them as: 

```fortran
use stdlib_constants, only: clight => c
```

Of course, it's important that their names are not confusing as @awvwgk is suggesting. But here they're taken from Scipy, so, probably a good reference for most people coming from the Python world.

Reply:
I am in favour of keeping the same notation as in Scipy.
@MilanSkocic Could you maybe add a warning in the specs, to highlight this potential issue and how to solve it (@perazz 's suggestion), please?

Reply:
Ok, no problem.

Reply:
Done.